src/UI/Settings/ProviderSettingsModelBase.js

```diff
@@ -3,6 +3,66 @@ var DeepModel = require('backbone.deepmodel');
 var Messenger = require('../Shared/Messenger');
 
 module.exports = DeepModel.DeepModel.extend({
+    connectData : function(action) {
+        var self = this;
+        
+        this.trigger('connect:sync');
+
+        var promise = $.Deferred();
+
+        var callAction = function(action) {
+            var params = {};
+            params.url = self.collection.url + '/connectData/' + action;
+            params.contentType = 'application/json';
+            params.data = JSON.stringify(self.toJSON());
+            params.type = 'POST';
+            params.isValidatedCall = true;
+
+            $.ajax(params).fail(promise.reject).success(function(response) {
```

Comment:
`var promise = $.ajax(options);`

then use `promise.done()` and `promise.fail()` instead.